### PR TITLE
hello

### PR DESCRIPTION
**Pull request by nusjahan**

### CODE REVIEW

File: src/main/java/bank/connect/tech/model/Customer.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
+package bank.connect.tech.model;
+
+import javax.persistence.*;
+import java.util.Set;
+
+@Entity
+public class CustomerPOJO {
```
Comment (zmazanv) [inline src/main/java/bank/connect/tech/model/Customer.java]: This should be renamed  to Customer to match the filename.

File: src/main/java/bank/connect/tech/model/Customer.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    @Id
+   // @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    private String first_name;
+    private String last_name;
+    @OneToOne
```
Comment (zmazanv) [inline src/main/java/bank/connect/tech/model/Customer.java]: This should be `@OneToMany` rather than `@OneToOne` considering that each customer can have multiple addresses, as well as this is a Set, by which it holds many of a type of object, being of the type Address in this case.

File: src/main/java/bank/connect/tech/model/Customer.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    private Set<Address> addresses;
+
+
+    public CustomerPOJO() {
```
Comment (zmazanv) [inline src/main/java/bank/connect/tech/model/Customer.java]: You can remove the constructor as it is not necessary. Jackson uses the nullary constructor, along with setters, to create the dependency objects. So it can be omitted.

File: src/main/java/bank/connect/tech/model/Customer.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    }
+
+    public CustomerPOJO(Long id, String first_name, String last_name, Set<String> addresses) {
```
Comment (zmazanv) [inline src/main/java/bank/connect/tech/model/Customer.java]: Once again, like with the other constructor, this constructor is not necessary as Jackson just uses the setters and nullary constructor to instantiate dependency objects. It can be removed.

**Review by ev3lasco — APPROVED** [review]

File: src/main/java/bank/connect/tech/repo/CustomerRepo.java [hunk 5]
```diff
@@ -0,0 +1,2 @@
+package bank.connect.tech.repo;public interface CustomerRepo {
```
Comment (zmazanv) [inline src/main/java/bank/connect/tech/repo/CustomerRepo.java]: I would suggest putting these on separate lines to improve readability, as well as renaming it to `CustomerRepository` to match the others. Also ensure it extends CrudRepository. It would look like would look like the following:

``` Java
package bank.connect.tech.repo;

import bank.connect.tech.model.Customer;
import org.springframework.data.repository.CrudRepository;

public inteface CustomerRepository extends CrudRepository<Customer, Long> {}
 ```

File: src/main/java/bank/connect/tech/model/Customer.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
+package bank.connect.tech.model;
+
+import javax.persistence.*;
+import java.util.Set;
+
+@Entity
+public class CustomerPOJO {
+
+    @Id
+   // @GeneratedValue(strategy = GenerationType.IDENTITY)
```
Comment (zmazanv) [inline src/main/java/bank/connect/tech/model/Customer.java]: This can be removed, as it is not needed.

File: src/main/java/bank/connect/tech/controller/CustomerController.java [hunk 7]
```diff
@@ -0,0 +1,2 @@
+package bank.connect.tech.controller;public class CustomerController {
```
Comment (zmazanv) [inline src/main/java/bank/connect/tech/controller/CustomerController.java]: I would suggest putting these on separate lines to improve readability. It would look like the following:

``` Java
package bank.connect.tech.controller;

public class CustomerController {
}
```

File: src/main/java/bank/connect/tech/Service/CustomerService.java [hunk 8]
```diff
@@ -0,0 +1,2 @@
+package bank.connect.tech.Service;public class CustomerService {
```
Comment (zmazanv) [inline src/main/java/bank/connect/tech/Service/CustomerService.java]: I would suggest putting these on separate lines to improve readability. It would look like the following:

``` Java
package bank.connect.tech.controller;

public class CustomerService {
}
```

**Review by zmazanv — CHANGES_REQUESTED** [review]
Please review the changes. Good start. 😀

File: src/main/java/bank/connect/tech/model/Customer.java [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+    private String last_name;
+    @OneToMany
+    @Column
+    @JoinColumn(name = "customer_id", nullable = false)
+    @OnDelete(action = OnDeleteAction.CASCADE)
+    @JsonIgnore
+    private Set<Address> addresses;
+
+
+    public Customer() {
```
Comment (zmazanv) [inline src/main/java/bank/connect/tech/model/Customer.java:27]: Please remove this constructor as it is not needed. Jackson, which is the library that Spring uses to create dependency objects, uses the setters to create needed objects, hence this is not needed.

File: src/main/java/bank/connect/tech/model/Customer.java [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+    }
+
+    public Customer(Long id, String first_name, String last_name, Set<Address> addresses) {
```
Comment (zmazanv) [inline src/main/java/bank/connect/tech/model/Customer.java:30]: Please remove this constructor as it is not needed. Jackson, which is the library that Spring uses to create dependency objects, uses the setters to create needed objects, hence this is not needed.

File: src/main/java/bank/connect/tech/repo/CustomerRepo.java [hunk 11]
```diff
@@ -0,0 +1,11 @@
+package bank.connect.tech.repo;
+
+import bank.connect.tech.model.Customer;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface CustomerRepo extends CrudRepository<Customer, Long> {
```
Comment (zmazanv) [inline src/main/java/bank/connect/tech/repo/CustomerRepo.java:8]: Please refactor and rename the interface to be CustomerRepository so that it matches our other repositories.

File: src/main/java/bank/connect/tech/repo/CustomerRepo.java [hunk 12]
```diff
@@ -0,0 +1,11 @@
...
+
```
Comment (zmazanv) [inline src/main/java/bank/connect/tech/repo/CustomerRepo.java:9]: Additionally please remove lines 9 and 10 so we don't have unnecessary whitespace.

**Review by zmazanv — CHANGES_REQUESTED** [review]

**Review by zmazanv — APPROVED** [review]
This all looks good. We can continue.
